lib/src/server/server.dart

```diff
@@ -117,8 +122,11 @@ class Server {
         handler?.cancel();
         _connections.remove(connection);
       });
-    }, onError: (error) {
-      print('Socket error: $error');
+    }, onError: (error, stackTrace) {
+      if (error is Error) {
+        Zone.current.handleUncaughtError(error, stackTrace);
+      }
+      // Non-Errors are ignored by the server.
```

Comment:
What exceptions could happen?

Reply:
I'm not sure what could actually happen. accept(2) on Linux does have some odd errors that I'm not sure about, but most of them seem unlikely to happen for Dart because it's managed. ECONNABORTED seems super weird and unspecified when it can happen, I'd have to investigate that. The Linux manpage also seems to imply that accept(2) might receive an socket error for the accepted socket if there was one pending, also a bit surprising. Then there's the question of what errors can happen on the dart side. Nevertheless I've never seen any accept() loop actually handle such possibilities and just die when they can't accept a socket.

My gut feeling is that you should probably just die on any exception. If you find that ever happens spuriously, I'd be interested in that condition.

Reply:
https://github.com/dart-lang/sdk/blob/5721d8af6d8755fa6f70f3951de2c451c57bf12f/sdk/lib/_http/http_impl.dart#L2582 seems to ignore HandshakeException.